scripts/translateWithAi.ts

```diff
@@ -0,0 +1,513 @@
+import { Option, Result } from "@swan-io/boxed";
+import chalk from "chalk";
+import cliSpinners from "cli-spinners";
+import fs from "fs/promises";
+import { ChatCompletionRequestMessage, Configuration, OpenAIApi } from "openai";
+import ora from "ora";
+import os from "os";
+import path from "path";
+import prompts from "prompts";
+import tiktoken from "tiktoken-node";
+import type { Except } from "type-fest";
+
+/**
+ * This script is used to translate the app using OpenAI's GPT-3.5 API.
+ * - First we list all the translations we want to run with tuples of [app, locale]
+ * - For each tuple, we generate a prompt
+ *   - this prompt contains as context some translated keys with the translation we use. (This explain to openai how we translated other keys and help it to generate consistent translations)
+ *   - creating a context is just a list of message where we can say "When I ask you this, you should answer this"
+ * - Once the prompt is generated, we compute a price approximation
+ *   - first we count the number of tokens in the prompt (because proce is based on number of tokens) https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them
+ *     (it uses tiktoken-node, after a few test results wasn't 100% accurate, but it's close enough for approximating the price)
+ *   - then we apply openai api price per token and display a prompt to confirm if we want to call openai api
+ * - If the user confirms, we call openai api and wait for the result with a spinner because it can take a while (~1 second per key to translate)
+ * - Once we have the result, we parse it and edit the translation file
+ */
+
+const OPENAI_API_KEY = process.env.OPENAI_API_KEY;
```

Comment:
You cannot use `.env` file, must use `.env.build` for this (https://github.com/swan-io/swan-partner-mobile/blob/main/.env.build.example) as `.env` values will be embedded in the app and could be retrieve using decompilation.